rest/tlsconfig/reloader.go

```diff
@@ -2,83 +2,80 @@ package tlsconfig
 
 import (
 	"crypto/tls"
+	"os"
 	"sync"
 	"sync/atomic"
 	"time"
 
-	"github.com/fsnotify/fsnotify"
-	"github.com/go-phorce/dolly/algorithms/throttle"
 	"github.com/juju/errors"
 )
 
+// Wrap time.Tick so we can override it in tests.
+var makeTicker = func(interval time.Duration) (func(), <-chan time.Time) {
+	t := time.NewTicker(interval)
+	return t.Stop, t.C
+}
+
 // KeypairReloader keeps necessary info to provide reloaded certificate
 type KeypairReloader struct {
-	watcher    *fsnotify.Watcher
-	lock       sync.RWMutex
-	loadedAt   time.Time
-	count      uint32
-	keypair    *tls.Certificate
-	certPath   string
-	keyPath    string
-	throttle   throttle.Driver
-	inProgress bool
+	lock           sync.RWMutex
+	loadedAt       time.Time
+	count          uint32
+	keypair        *tls.Certificate
+	certPath       string
+	certModifiedAt time.Time
+	keyPath        string
+	keyModifiedAt  time.Time
+	inProgress     bool
+	stopChan       chan<- struct{}
 }
 
-const evtToWatch = fsnotify.Write
-
 // NewKeypairReloader return an instance of the TLS cert loader
-func NewKeypairReloader(certPath, keyPath string, throttlePeriod time.Duration) (*KeypairReloader, error) {
-	watcher, err := fsnotify.NewWatcher()
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-
+func NewKeypairReloader(certPath, keyPath string, checkInterval time.Duration) (*KeypairReloader, error) {
 	result := &KeypairReloader{
-		watcher:  watcher,
 		certPath: certPath,
 		keyPath:  keyPath,
+		stopChan: make(chan struct{}),
 	}
 
 	logger.Infof("api=NewKeypairReloader, status=started")
 
-	err = result.Reload()
+	err := result.Reload()
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
 
-	result.throttle = throttle.Func(throttlePeriod, true, func() {
-		if err := result.Reload(); err != nil {
-			logger.Errorf("api=NewKeypairReloader, reason=reload, cert='%s', key='%s', err=[%v]", certPath, keyPath, errors.ErrorStack(err))
-		}
-	})
-
+	stopChan := make(chan struct{})
+	tickerStop, tickChan := makeTicker(checkInterval)
 	go func() {
-		closed := false
-		for !closed {
+		for {
 			select {
-			case event, ok := <-watcher.Events:
-				if !ok {
-					closed = true
-				}
-				if event.Op&evtToWatch != 0 {
-					logger.Noticef("api=NewKeypairReloader, reason=event, event=[%v]", event)
-					result.throttle.Trigger()
+			case <-stopChan:
+				tickerStop()
+				logger.Infof("api=NewKeypairReloader, status=closed, count=%d", result.LoadedCount())
+				return
+			case <-tickChan:
+				modified := false
+				fi, err := os.Stat(certPath)
+				if err == nil {
```

Comment:
I wonder if you wanna log this error instead of just ignoring? Otherwise modified might stay false infinitely without ever logging any error.